### DISCUSSION TITLE
Resampling always showing Nearest although other options are used

### DISCUSSION THREAD

**Discussion by lawrencenika** [developmentseed/titiler#981] [Q&A]
#### Problem description

I tried with `/cog/tiles/WebMercatorQuad/{z}/{x}/{y}.png?url=${encodeURIComponent(objPath)}&bidx=1&resampling=bilinear` and a bunch of other resampling options like `gauss` or `cubic` but when zooming in to the image, they all look pixelated still which is from default `resampling=nearest`. Anything else I can try?

what i m seeing is
![image](https://github.com/user-attachments/assets/3313e1c2-3564-42d9-b7a2-5a6efe09ccb6)
when `resampling=bilinear`

#### Expected Output

![image](https://github.com/user-attachments/assets/3ad45993-c5d0-4931-926c-249ffbc85a31)
for `resampling=bilinear`

#### Environment Information

titiler:latest (v0.18.6) run in local docker env on macos
with ENV WORKERS_PER_CORE 1

**Comment by vincentsarago** [comment]
@lawrencenika would you be able to share the file you're working with?

**Reply by lawrencenika** [reply]
yes sure, https://drive.google.com/file/d/1bZQAvxEBbPMyjkXhNYEVyQQck0dNMFRx/view?usp=sharing

**Reply by lawrencenika** [reply]
follow up on the above if there is a way to make it work @vincentsarago

**Reply by vincentsarago** [reply]
@lawrencenika there are 2 resampling at place:
- dataset reprojection 
- dataset reading 

each one are controlled by specific options: `resampling` and `reproject`. 

you can try with `resampling=bilinear&reproject=bilinear`

**Reply by lawrencenika** [reply]
do I have to use them together? when I tried with `resampling` only, it failed

**Reply by vincentsarago** [reply]
what failed? 

both control different things as I've explained. you can use them independently!

**Reply by lawrencenika** [reply]
when I tried with `&bidx=1&resampling=bilinear` it still returns the same pixelated image when zoom in, it might be my misconception but when I tried linear resampling in local desktop software for the same image, it returns the smooth image when zoomed in (shown in expected output section).

**Reply by vincentsarago** [reply]
>  tried linear resampling in local desktop software for the same image, it returns the smooth image when zoomed in (shown in expected output section).

Your local software might apply resampling after dataset read, which we don't support. 


You can clearly see the effect of `resampling` when using the `/preview` endpoint 

`rescale=5000,11000&resampling=nearest&bidx=1&width=2048&height=2048`

![neartest](https://github.com/user-attachments/assets/0b9518ed-f7f4-401c-af63-8e2b8c41508c)

`rescale=5000,11000&resampling=cubic&bidx=1&width=2048&height=2048`

![cubic](https://github.com/user-attachments/assets/f5fa3d88-4a5a-4824-be0c-7c5c6c8d1eaf)


The `resampling` as a bit less effect when using the `/tiles` endpoint because you're doing reprojection on small chips which I why you should try seting `reproject=` but still you won't get the same result as you're get from you're local software (which is?)
